storage/innobase/xtrabackup/src/fil_cur.cc

```diff
@@ -57,20 +57,7 @@ static bool xb_get_zip_size(const char *file_name, pfs_os_file_t file,
     page_size.copy_from(page_size_t(flags));
   }
 
-  if (page_size.is_compressed() || FSP_FLAGS_GET_ENCRYPTION(flags)) {
-    is_compressable = false;
-  } else {
-    const auto ret = os_file_read(read_request, file_name, file, buf, 0,
-                                  page_size.physical() * 2);
-    if (!ret) {
-      return (false);
-    }
-    if (Compression::is_compressed_page(buf + page_size.physical())) {
-      is_compressable = false;
-    } else {
-      is_compressable = true;
-    }
-  }
+  is_compressable = FSP_FLAGS_GET_ENCRYPTION(flags) ? false : true;
```

Comment:
A tablespace can be encrypted and PageIOcompressed at same time. will this avoid doing punch holes on such tablespaces?

Reply:
encrypted tables are not compressed during backup. That is what this part of code validates (if this table will be entitle to be compressed or not).

For example, we test it at https://github.com/altmannmarcelo/percona-xtrabackup/blob/PXB-2890-8.0/storage/innobase/xtrabackup/test/inc/page_compression_common.sh#L35-L38 that gets called by https://github.com/altmannmarcelo/percona-xtrabackup/blob/PXB-2890-8.0/storage/innobase/xtrabackup/test/suites/compression/page_compression_zstd.sh